eth/backend.go

```diff
@@ -199,8 +200,11 @@ func New(stack *node.Node, config *ethconfig.Config, l1Client sync_service.EthCl
 		return nil, err
 	}
 	if config.CheckCircuitCapacity {
-		tracer := tracing.NewTracerWrapper()
-		eth.blockchain.Validator().SetupTracerAndCircuitCapacityChecker(tracer)
+		eth.asyncChecker = ccc.NewAsyncChecker(eth.blockchain, runtime.GOMAXPROCS(0))
+		eth.asyncChecker.WithOnFailingBlock(func(b *types.Block, err error) {
+			log.Warn("block failed CCC check, it will most likely be reorged by the sequencer", "hash", b.Hash(), "err", err)
+		})
+		eth.blockchain.Validator().WithAsyncValidator(eth.asyncChecker.Check)
```

Comment:
Minor nitpick: `.With...` gives me the impression that this method will return a new copy instead of modifying the original object.

```suggestion
		eth.blockchain.Validator().RegisterAsyncValidator(eth.asyncChecker.Check)
```

Reply:
this is actually a variation of [FOP](https://golang.cafe/blog/golang-functional-options-pattern.html), but I don't mind renaming.